api/v1alpha1/httpchaos_types.go

```diff
@@ -0,0 +1,238 @@
+// Copyright 2020 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// KindHTTPChaos is the kind for http chaos
+const KindHTTPChaos = "HTTPChaos"
+
+func init() {
+	all.register(KindHTTPChaos, &ChaosKind{
+		Chaos:     &HTTPChaos{},
+		ChaosList: &HTTPChaosList{},
+	})
+}
+
+// HTTPAction represents the chaos action about I/O action.
+type HTTPChaosAction string
+
+const (
+	HTTPDelayAction HTTPChaosAction = "delay"
+	HTTPAbortAction                 = "abort"
+	HTTPMixedAction                 = "mixed"
+)
+
+type Matcher struct {
+	Name           string  `json:"name"`
+	ExactMatch     *string `json:"exact_match,omitempty"`
+	RegexMatch     *string `json:"regex_match,omitempty"`
+	SafeRegexMatch *string `json:"safe_regex_match,omitempty"`
+	RangeMatch     *string `json:"range_match,omitempty"`
+	PresentMatch   *string `json:"present_match,omitempty"`
+	PrefixMatch    *string `json:"prefix_match,omitempty"`
+	SuffixMatch    *string `json:"suffix_match,omitempty"`
+	InvertMatch    *string `json:"invert_match,omitempty"`
```

Comment:
We don't need to realize the whole `Match` in the next requests. I think `Exact`/ `Prefix`/ `Suffix` is enough.

Reply:
The whole `Match` was implement by the envoy . We just need to send the string to envoy , so I think keeping the whole `Match` is better. 
And I have keep a detailed document in crd description offered by envoy,which is easy to understand.
 [https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-headermatcher](url)